### PR TITLE
attempt to improve review experience using GHA workflow

### PR DESCRIPTION
**Pull request by hansvancalster**
## Description

Attempt to write a GitHub action workflow that generates rendered version of a protocol as an artifact of a pull request.
I have no idea if this will work.... I used code snippets from https://github.com/inbo/tutorials/tree/master/.github/workflows and https://github.com/r-lib/actions/blob/master/examples/render-rmarkdown.yaml.

The idea is to only render the protocol that is in review. The tricky part is how to specify in the workflow where the "index.Rmd" is located. I am really not sure if the RMD_PATH specification in the workflow will do what I think it does.

## Related Issue

See #25

### CODE REVIEW

**Comment by ThierryO** [comment]
I've solved a similar problem: [R code](https://github.com/inbo/indicatoren/blob/00e68ace2e4810fb452e82b414dc3254fc68e943/render_functions.R#L460), [GitHub Action yml]() and [custom GitHub Action](https://github.com/inbo/actions/tree/master/render_indicator)

**Comment by hansvancalster** [comment]
Thanks. I'll have a look. Can you fix the second link in your post (it links to this PR) (or maybe it just needs to be removed)?

**Comment by ThierryO** [comment]
The link is fixed.

**Comment by hansvancalster** [comment]
@ThierryO do you see why the github action is not running after my last push. In theory it should run on pull_request (default triggers are = opened, synchronize, reopened). Pushing a commit to the branch causes synchronize, so it should run?

**Comment by ThierryO** [comment]
It actually did try run but [failed](https://github.com/inbo/protocols/actions/runs/766383046)

File: .github/workflows/generate-artifact-pr.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Install packages
+        run: |
+          R -e 'install.packages("remotes")'
+          R -e 'remotes::install_github("protocolhelper")
+
+      - name: Render protocol (only the one that is the focus of the PR) 
+        run: |
+          mkdir -p artifact
+          Rscript -e 'protocolhelper::render_protocol(${{ GITHUB_HEAD_REF }}, "artifact")'
```
Comment (ThierryO) [inline .github/workflows/generate-artifact-pr.yml:33]:
```suggestion
          Rscript -e 'protocolhelper::render_protocol(${{ github.event.pull_request.head.sha }}, "artifact")'
```

File: .github/workflows/REVIEWING.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Reviewing pull requests
+
+Thanks to GitHub Actions, an artifact (=zip file) of the rendered protocol is now automatically created on each pull request.
+
+Reviewers can proceed like that:
```
Comment (ThierryO) [inline .github/workflows/REVIEWING.md:5]:
```suggestion
Reviewers can proceed like this:
```

File: .github/workflows/REVIEWING.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+1) On the PR page, you can find a "details" link under "checks - On PR, build the site and ...". Go there, click on the top link in the left sidebar, and download the generated artifact at the bottom of the page.
+2) Decompress it and make sure the target directory is called 'protocols' (you may need to rename it)
```
Comment (ThierryO) [inline .github/workflows/REVIEWING.md:8]:
```suggestion
2) Unzip it and make sure the target directory is called 'protocols' (you may need to rename it)
```

File: .github/workflows/REVIEWING.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+3) From the parent directory (just above tutorials), run `python -m http.server 8887`, _or_ launch the Google Chrome [Web Server app](https://chrome.google.com/webstore/detail/web-server-for-chrome/ofhbbkphhbklhfoeikjpcbhemlocgigb) and point it at the parent directory.
```
Comment (ThierryO) [inline .github/workflows/REVIEWING.md:9]: This seems complicated.  Why not use the standard GitHub tools?

File: .github/workflows/REVIEWING.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+4) Point your browser to http://localhost:8887/tutorials.
+5) Review the updated website and accept / refuse / comment the PR
+
+**Note: for step 3, you can use any other simple HTTP server to serve the current directory if you don't have a Python 3 environment or Google Chrome available.**
```
Comment (ThierryO) [inline .github/workflows/REVIEWING.md:13]: What if I have no clue what you are talking about? Or I have neither?

File: .github/workflows/REVIEWING.md (hunk 4 above)
Comment (hansvancalster) [reply .github/workflows/REVIEWING.md:9]: I need to simplify this. This is copy pasted from the tutorials repo (hugo website) and can be much simplified.

File: .github/workflows/REVIEWING.md (hunk 5 above)
Comment (hansvancalster) [reply .github/workflows/REVIEWING.md]: See 
458044b

File: .github/workflows/REVIEWING.md (hunk 4 above)
Comment (hansvancalster) [reply .github/workflows/REVIEWING.md]: see 
458044b

**Pull request merged**